teammapper-frontend/mmp/src/map/handlers/nodes.ts

```diff
@@ -640,6 +641,77 @@ export default class Nodes {
         }
     }
 
+    /**
+     * Proactively apply coordinates to all nodes that are missing coordinates
+     * This method is very similar to calculateCoordinates, but it's fully self-contained and only applicable to MapSnapshots.
+     * Adding this functionality to calculateCoordinates would break a lot of things, as Node and ExportNodeProperties are not the same.
+     * @param {MapSnapshot} map
+     * @returns {MapSnapshot} map - Return the map snapshot with properly applied coordinates
+     */
+    public applyCoordinatesToMapSnapshot(map: MapSnapshot): MapSnapshot {
+        const getOrientation = (node: ExportNodeProperties, rootNode: ExportNodeProperties) => node.coordinates?.x < rootNode.coordinates?.x
+        const getLowerNode = (nodes: ExportNodeProperties[]) => {
+            let tmp = nodes[0].coordinates?.y || 0
+            let lowerNode = nodes[0]
+
+            nodes.map(node => {
+                if (node.coordinates && node.coordinates.y > tmp) {
+                    tmp = node.coordinates.y
+                    lowerNode = node
+                }
+            })
+
+            return lowerNode
+        }
+
+        const rootNode = map.find(x => x.isRoot)
+
+        map.map(node => {
+            if (!node.coordinates) {
+                const nodeParent = node.parent ? map.find(x => x.id === node.parent) : null
+                let coordinates: Coordinates = {
+                    x: nodeParent ? nodeParent.coordinates?.x : 0,
+                    y: nodeParent ? nodeParent.coordinates?.y : 0
+                }
+    
+                let siblings = node.parent ? map.filter(x => x.parent === node.parent && x.id !== node.id) : null
+                
+                if (nodeParent && nodeParent.isRoot) {
+                    const rightNodes: Array<ExportNodeProperties> = [],
+                        leftNodes: Array<ExportNodeProperties> = []
+
+                    siblings.map(sibling => {
+                        getOrientation(sibling, rootNode) ? leftNodes.push(sibling) : rightNodes.push(sibling)
+                    })
+
+                    if (leftNodes.length <= rightNodes.length) {
+                        coordinates.x -= 200
```

Comment:
magic numbers should be constants, that you put at the top of the file or somewhere

Reply:
These are now defined as constants. I've also applied these constants to `calculateCoordinates`.